components/org.wso2.carbon.identity.organization.management.service/src/main/java/org/wso2/carbon/identity/organization/management/service/OrganizationManager.java

```diff
@@ -368,4 +369,14 @@ default List<String> getOrganizationsMetaAttributes(Integer limit, String after,
         throw new OrganizationManagementServerException("getOrganizationsMetaAttributes is not implemented in "
                 + this.getClass().getName());
     }
+
+    /**
+     * Retrieves a map of organization IDs to their corresponding organization names.
+     *
+     * @param orgIdList The list of organization IDs to retrieve names for.
+     * @return A map where each key is an organization ID and the corresponding value is the organization name.
+     *         If the input list is null or empty, an empty map will be returned.
+     * @throws OrganizationManagementException If an error occurs while retrieving organization names for the given IDs.
+     */
+    Map<String, String> getOrganizationIdToNameMap(List<String> orgIdList) throws OrganizationManagementException;
```

Comment:
Shall we pass the parent Org id or domain name for this method to filter the org names respective to the given parent org only.
Otherwise anyone can use this method to get the org name of any organization where it may have the access to the information of other tenant orgs in a shared environment.

Reply:
Since this OSGi method's usage is from another OSGi service which is developed by a developer, and this is a meta data retreiveal no need to add unneccasery validation like requested org list is under a API invoked tenant domain.
This validation should have been validated from API layer (through valves) or data storage

Reply:
But since this is a public interface isn't it better to provide it in a restricted way ?